source/server/listener_impl.h

```diff
@@ -156,9 +158,9 @@ class PerListenerFactoryContextImpl : public Configuration::ListenerFactoryConte
                                 ProtobufMessage::ValidationVisitor& validation_visitor,
                                 const envoy::config::listener::v3::Listener& config_message,
                                 const Network::ListenerConfig* listener_config,
-                                ListenerImpl& listener_impl, DrainManagerPtr drain_manager)
+                                ListenerImpl& listener_impl, DrainManagerSharedPtr drain_manager)
       : listener_factory_context_base_(std::make_shared<ListenerFactoryContextBaseImpl>(
-            server, validation_visitor, config_message, std::move(drain_manager))),
+            server, validation_visitor, config_message, drain_manager)),
```

Comment:
I think std::move was still appropriate here, it avoids changes to the shared_ptr refcount